### ISSUE TITLE
Missing header in load_balancer_pool

### ISSUE THREAD

**Issue by DDuarte** [cloudflare/terraform-provider-cloudflare#957]
### Terraform version

```
Terraform v0.14.2
+ provider registry.terraform.io/cloudflare/cloudflare v2.17.0
```

### Affected resource(s)

- load_balancer_pool

### Expected behavior

Host header can be overriden in `load_balancer_pool`

https://api.cloudflare.com/#load-balancer-pools-create-pool

```json
[
  {
    "name": "app-server-1",
    "address": "0.0.0.0",
    "enabled": true,
    "weight": 0.56,
    "header": {  <---
      "Host": [
        "example.com"
      ]
    }
  }
]
```

### Actual behavior

`header` attribute is missing from the `load_balancer_pool` resource

**Comment by stoilis** [comment]
I can confirm that this indeed doesn't work, but not sure if it's a bug or a feature request. The documentation doesn't mention the headers as a valid attribute https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/load_balancer_pool

**Comment by jacobbednarz** [comment]
not a bug; it looks to be a new field seeing how the `cloudflare-go` library also doesn't have it (and will need it before Terraform can use it) https://github.com/cloudflare/cloudflare-go/blob/4f732091c7c5796a1957288aff22bf90979fe42d/load_balancing.go#L29-L34

**Comment by alam0rt** [comment]
I have opened a PR https://github.com/cloudflare/cloudflare-go/pull/603

Hopefully we can get this in ASAP as I need to override the host header myself

**Comment by alam0rt** [comment]
Here we go: https://github.com/cloudflare/cloudflare-go/blob/8f701f8f9c7bfa0326d10f90c599b9dc72fc4e89/load_balancing.go#L36

**Comment by xenalite** [comment]
@jacobbednarz Does this require a release of another version of cloudflare-go before it can be consumed in this repo?

**Comment by jacobbednarz** [comment]
Yes

**Comment by quinndiggity** [comment]
@xenalite @jacobbednarz it has been released https://github.com/cloudflare/cloudflare-go/releases/tag/v0.16.0

https://github.com/cloudflare/cloudflare-go/blob/v0.16.0/load_balancing.go#L36

**Comment by DDuarte** [comment]
Fixed by #1042
